llvm/include/llvm/CodeGen/SelectionDAGNodes.h

```diff
@@ -652,7 +652,7 @@ END_TWO_BYTE_PACK()
   SDUse *UseList = nullptr;
 
   /// The number of entries in the Operand/Value list.
-  unsigned short NumOperands = 0;
+  uint32_t NumOperands = 0;
```

Comment:
If I counted bytes right, I believe this increases the side of SDNode by 8 bytes on 64-bit targets. This was perfectly paired with the NumValues and NodeId to use up exactly 8 bytes. Now it uses 10 bytes and leaves a 6 byte hole before ValueList.

We might be able to gain back some bytes by shrinking SDNodeFlags to 16 bits and rearranging some fields.